diff --git a/LibreTransmitter/LibreTransmitterUI/Views/Settings/SettingsView.swift b/LibreTransmitter/LibreTransmitterUI/Views/Settings/SettingsView.swift
index ac1b6e1..3e0434b 100644
--- a/LibreTransmitter/LibreTransmitterUI/Views/Settings/SettingsView.swift
+++ b/LibreTransmitter/LibreTransmitterUI/Views/Settings/SettingsView.swift
@@ -118,17 +118,17 @@ struct SettingsView: View {
                 measurementSection
                 if let date = glucoseMeasurement.predictionDate, let prediction = glucoseMeasurement.prediction {
                     Section(header: Text(LocalizedString("Last Blood Sugar prediction", comment: "Text describing header for Blood Sugar prediction section"))) {
-                        SettingsItem(title: "CurrentBG", detail: displayGlucosePreference.format(prediction))
-                        SettingsItem(title: "Date", detail: longDateFormatter.string(from: date) )
+                        SettingsItem(title: String(localized: "CurrentBG"), detail: displayGlucosePreference.format(prediction))
+                        SettingsItem(title: String(localized: "Date"), detail: longDateFormatter.string(from: date) )
                     }
                 }
                 
                 NavigationLink(destination: deviceInfoSection) {
-                    SettingsItem(title: "Device details")
+                    SettingsItem(title: String(localized: "Device details"))
                 }
                 
                 NavigationLink(destination: CalibrationEditView()) {
-                    Button(Features.allowsEditingFactoryCalibrationData ? "Edit calibrations" : "View factory calibrations") {
+                    Button(Features.allowsEditingFactoryCalibrationData ? String(localized: "Edit calibrations") : String(localized: "View factory calibrations")) {
                         print("edit calibration clicked")
                     }
                 }
@@ -168,9 +168,9 @@ struct SettingsView: View {
         }
 
         return Section(header: Text(LocalizedString("Last measurement", comment: "Text describing header for last measurement section"))) {
-            SettingsItem(title: "Glucose", detail: glucoseText)
-            SettingsItem(title: "Date", detail: glucoseDateText )
-            SettingsItem(title: "Sensor Footer checksum", detail: $glucoseMeasurement.checksum )
+            SettingsItem(title: String(localized: "Glucose"), detail: glucoseText)
+            SettingsItem(title: String(localized: "Date"), detail: glucoseDateText )
+            SettingsItem(title: String(localized: "Sensor Footer checksum"), detail: $glucoseMeasurement.checksum )
         }
     }
 
@@ -178,42 +178,42 @@ struct SettingsView: View {
         List {
             Section(header: Text(LocalizedString("Device Info", comment: "Text describing header for device info section"))) {
                 if !transmitterInfo.battery.isEmpty {
-                    SettingsItem(title: "Battery", detail: $transmitterInfo.battery )
+                    SettingsItem(title: String(localized: "Battery"), detail: $transmitterInfo.battery )
                 }
                 
                 // The firmware version is not always extractable for all devices
                 // and the libre2 direct version does not support it at all
                 if !transmitterInfo.hardware.isEmpty {
-                    SettingsItem(title: "Hardware", detail: $transmitterInfo.hardware )
+                    SettingsItem(title: String(localized: "Hardware"), detail: $transmitterInfo.hardware )
                 }
                 // The firmware version is not always extractable for all devices
                 // and the libre2 direct version does not support it at all
                 if !transmitterInfo.firmware.isEmpty {
-                    SettingsItem(title: "Firmware", detail: $transmitterInfo.firmware )
+                    SettingsItem(title: String(localized: "Firmware"), detail: $transmitterInfo.firmware )
                 }
                 
-                SettingsItem(title: "Connection State", detail: $transmitterInfo.connectionState )
-                SettingsItem(title: "Transmitter Type", detail: $transmitterInfo.transmitterType )
+                SettingsItem(title: String(localized: "Connection State"), detail: $transmitterInfo.connectionState )
+                SettingsItem(title: String(localized: "Transmitter Type"), detail: $transmitterInfo.transmitterType )
                 
                 // The mac address of a given device is normally not available on ios
                 // Only the bluetooth identifier, which is a normalized derivative of the mac address is available
                 // However, some transmitters, such as the bubble, provide their own mac address as part of its advertisement info
                 // which we extract and put herer
                 if !transmitterInfo.transmitterMacAddress.isEmpty {
-                    SettingsItem(title: "Mac", detail: $transmitterInfo.transmitterMacAddress )
+                    SettingsItem(title: String(localized: "Mac"), detail: $transmitterInfo.transmitterMacAddress )
                 }
                 
-                SettingsItem(title: "Sensor Type", detail: $transmitterInfo.sensorType )
+                SettingsItem(title: String(localized: "Sensor Type"), detail: $transmitterInfo.sensorType )
                 
-                SettingsItem(title: "Sensor Start", detail: sensorInfo.activatedAtString )
-                SettingsItem(title: "Sensor End", detail: sensorInfo.expiresAtString )
+                SettingsItem(title: String(localized: "Sensor Start"), detail: sensorInfo.activatedAtString )
+                SettingsItem(title: String(localized: "Sensor End"), detail: sensorInfo.expiresAtString )
             }
         }
         .textSelection(.enabled)
     }
 
     private var doneButton: some View {
-        Button("Done", action: {
+        Button(String(localized: "Done"), action: {
             notifyComplete.notify()
         })
     }
@@ -224,14 +224,14 @@ struct SettingsView: View {
             NavigationLink(destination: AuthView(completeNotifier: notifyComplete, notifyReset: notifyReset, notifyReconnect: notifyReconnect, pairingService: pairingService, bluetoothSearcher: bluetoothSearcher)) {
                 /*Button("Change Sensor") {
                 }.foregroundColor(.blue)*/
-                SettingsItem(title: "Change Sensor").foregroundColor(.blue)
+                SettingsItem(title: String(localized: "Change Sensor")).foregroundColor(.blue)
             }
         }
     }
 
     var destructSection: some View {
         Section {
-            Button("Delete CGM") {
+            Button(String(localized: "Delete CGM")) {
                 showingDestructQuestion = true
             }.foregroundColor(.red)
             .alert(isPresented: $showingDestructQuestion) {
@@ -262,21 +262,21 @@ struct SettingsView: View {
             // these subviews don't really need to be notified once glucose unit changes
             // so we just pass glucoseunit directly on init
             NavigationLink(destination: AlarmSettingsView(glucoseUnit: self.glucoseUnit)) {
-                SettingsItem(title: "Alarms")
+                SettingsItem(title: String(localized: "Alarms"))
             }
             
             if NotificationHelper.criticalAlarmsEnabled {
                 NavigationLink(destination: CriticalAlarmsVolumeView()) {
-                    SettingsItem(title: "Critical Alarms volume")
+                    SettingsItem(title: String(localized: "Critical Alarms volume"))
                 }
             }
             
             NavigationLink(destination: GlucoseSettingsView()) {
-                SettingsItem(title: "Glucose Settings")
+                SettingsItem(title: String(localized: "Glucose Settings"))
             }
         
             NavigationLink(destination: NotificationSettingsView()) {
-                SettingsItem(title: "Notifications")
+                SettingsItem(title: String(localized: "Notifications"))
             }
 
         }
@@ -443,9 +443,9 @@ struct SettingsView: View {
             }
             if sensorIsExpired {
                 VStack(alignment: .leading, spacing: 4) {
-                    Text("Sensor is expired")
+                    Text(String(localized: "Sensor is expired"))
                         .font(Font.subheadline.weight(.bold))
-                    Text("Replace sensor immediately to continue receving glucose values")
+                    Text(String(localized: "Replace sensor immediately to continue receving glucose values"))
                         .font(Font.footnote.weight(.semibold))
                 }.padding(.vertical, 8)
             } else if !["Notifying", "Connected"].contains(transmitterInfo.connectionState) || !showProgress {
